design-proposals/template-generation.md

```diff
@@ -0,0 +1,131 @@
+# Overview
+
+This design defines an approach to create Virtual Machine templates starting from an existing VM.
+
+## Motivation
+
+Creating a 'blueprint' of an existing VM to further tweak and use as a basis golden VM template is a common operation, also it is a tool that can be leveraged by UIs to improve the UX of the project.
+
+## Goals
+
+* Create a generic template of an existing VM.
```

Comment:
What is a `template` in Kubernetes?
Ain't there different template methods? Which of them is targeted by this proposal?

Reply:
Agreed. Maybe good to somehow hint at this by i.e. including a sentence around .. different mechanisms, because there is no default K8s way?

Reply:
I think there are existing template tools that could benefit from generic template API that would serve for metadata purposes. Examples I could think of or find are:
1. [Blog](https://livewyer.io/blog/2021/02/23/kubevirt-showcase-a-kubernetes-cluster-within-a-kubernetes-cluster/ )- Using Helm for templating VM (See [implementation](https://github.com/livewyer-ops/kubevirt-demo/tree/v1.0.0/examples/ubuntu/multi-vmis))
2. [Kustomize](https://kustomize.io/) - You could want to create a kustomize "template". You would pick a VM (Lets say you are using in testing stage for new app) and create a version with parameters better suited production deployment.

Ansible or mentioned Openshift templates (and other tools) could greatly benefit from this. The benefit I see is that there would be one API that knows what needs to be replaced or stay untouched.